gennet-module/gennet.py

```diff
@@ -203,24 +234,24 @@ def generate_network(ctx):
 def postprocess_network(G):
     G = nx.Graph(G)  # prune out parallel/multi edges
     G.remove_edges_from(nx.selfloop_edges(G))  # remove the self-loops
-    mapping = {i: f"{NODE_PREFIX}{ID_STR_SEPARATOR}{i}" for i in range(len(G))}
+    mapping = {i : f"{NODE_PREFIX}{ID_STR_SEPARATOR}{i}" for i in range(len(G))}
     return nx.relabel_nodes(G, mapping)  # label the nodes
 
 
 def generate_subnets(G, num_subnets):
```

Comment:
I would like some explanation about this function beacuse I don't fully understand it

Reply:
I just added handful of comments to the function. Do let me know if that helps. Recollect that  --num-subnets makes sure that the --num-nodes are grouped into as many subnets.

Reply:
I still don't got it.

If we assume that we have `num_subnets == 2`.
Assume we have 6 nodes like [4, 5, 3, 0, 2, 1]
Assume that `offsets == [1, 5]`

If we follow the for loop:
- end will starting as 1.
- In the inner loop, we will have `node-4` and `node-5` assigned to `subnetwork_0`
- Start will be `1`, end will be `5`
- Again, in the inner loop, `node-5` will be overwritten from `subnetwork_0` to `subnetwork_1`.

Is this behaviour desired?

Reply:
Yes, this is a desired behavior. To see why this must be so, consider a corner case like `lst == [5, 2, 1, 4, 3, 0]`,  `num_subnets == 4`, and   `offsets[3, 4, 5, 5] `. then without this overlap/overwriting (say, by doing `start = end + 1`), we would have one less subnet than what we require. The final result will be  `node-5, node-2, node-1, node-4  in subnetwork_0`, `node-3 in subnetwork_1` and `node-0 subnetwork_2`. That's 3 subnetworks as opposed 4. This overlap continuously   allows us to donate one node into subsequent subnets and hence the final subnetwork if needed.

This --num-subnets  overwrites, at each subnet boundary. The intervals are closed for a reason.